### PR TITLE
Support for deleting execution plans based on filters

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: lib/dynflow/persistence_adapters/sequel.rb [hunk 1]
```diff
@@ -36,13 +36,26 @@ def initialize(config)
...
+        filter(table(:execution_plan), filters).each_slice(batch_size) do |plans|
+          uuids = plans.map { |p| p.fetch(:uuid) }
+          @db.transaction do
+            table(:step).where(execution_plan_uuid: uuids).delete
+            table(:action).where(execution_plan_uuid: uuids).delete
+            count += table(:execution_plan).where(uuid: uuids).delete
+          end
+        end
+        return count
+      end
```
Comment (pitr-ch) [inline lib/dynflow/persistence_adapters/sequel.rb]: Isn't this missing abstract implementation in abstract persistence adapter?

Comment (iNecas) [inline lib/dynflow/persistence_adapters/sequel.rb]: Fixed

**Comment by iNecas** [comment]
Fixed comments and rebased: ready for merge (we need to get this in for #139, as there are changes in tests that would be quite handy there)

File: lib/dynflow/persistence_adapters/abstract.rb [hunk 2]
```diff
@@ -36,6 +36,10 @@ def find_execution_plans(options = {})
         raise NotImplementedError
       end
 
+      def delete_execution_plans(filters, batch_size = 1000)
+        raise NotImplementedError
+      end
```
Comment (pitr-ch) [inline lib/dynflow/persistence_adapters/abstract.rb]: What is the `filters` format? Could you add small example here or in the documentation?

**Comment by pitr-ch** [comment]
One doc-comment, ACK

File: lib/dynflow/persistence_adapters/abstract.rb (hunk 2 above)
Comment (iNecas) [inline lib/dynflow/persistence_adapters/abstract.rb]: updated

**Pull request merged**